### ISSUE TITLE
Concat Realm.Results

### ISSUE THREAD

**Issue by Fanda36** [realm/realm-js#2046]
## Goals

Concat Realm.Results of same types to new Realm.Results object.

## Expected Results

New Realm.Results where will be all entities and is it possible to apply filtered and others methods.

## Code Sample

```
res1 = contacts.filtered(q1)
contacts = contacts.filtered('!(' + q1 + ')')

res2 = contacts.filtered(q2)

out = res1.concat(res2)
outFiltered = out.filtered(q3)
```

**Comment by kneth** [comment]
[Realm Core](https://github.com/realm/realm-core) doesn't support concatenating results. You should use a combination of logical-or and logical-and: `contacts.filtered(`(${q1} || ${q2}) && ${q3}`).

**Comment by Fanda36** [comment]
I used combination (q1 || q2 || ...), it had best performance, but there was another problem. I need hold ordering (if I have q1 || q2, then I need have first time q1 results and then q2 results -> [q1_result, q1_result, .... last_q1_result, q2_result, ... last_q2_result]) - because for first time I need first name matches, then last name matches, then company matches... and there is not possible sort it by sorted method. I tried query (q1 || (q2 || (q3 ... ) ) ) too, but it was "randomly" ordered too.

So I found 2 solution, first is concat results from q1 and q2 as Array - performance is really good, but there are duplicitous and is really expansive to remove them:
``` javascript
out= Array.from(contacts.filtered(q1))
r2 = Array.from(contacts.filtered(q2))

// remove duplicites
r2 = r2.filter(function (n) {
   return out.findIndex((o) => n.id === o.id) === -1
});
out = out.concat(r2)
```

and second solution is get `contacts = contacts.filtered('!(' + q1 + ')') ` and use new `contacts` results it in next query and then concat everything like Array but it is slow solution too.
``` javascript
res1 = contacts.filtered(q1)
contacts = contacts.filtered('!(' + q1 + ')')
res2 = contacts.filtered(q2)
//  convert res1 and res2 to Array
//  and then concat
out = res1.concat(res2)
```
